### PR TITLE
CFEP 24: More Secure Package Name Approval Process

### PR DESCRIPTION
**Pull request by beckermr**
This PR proposes a CFEP for the new package name approval process.

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge/core

This PR falls under the CFEP Vote policy, please vote and/or comment on this PR. This PR needs 60% of core to vote yea to pass.

To vote please leave a PR review according to

 - Approve (yea)
 - Request Changes (nay)
 - Comment w/ "abstain" in it (abstain)

If you would like changes to the current language please leave a comment or push to this branch.

This vote will end on 2024-10-31.

**Review by beckermr — COMMENTED** [review]
I vote yae and am leaving a comment since I cannot leave an approve of my own PR.

**Review by ocefpaf — APPROVED** [review]

**Review by jaimergp — APPROVED** [review]

**Review by synapticarbors — APPROVED** [review]

**Review by isuruf — APPROVED** [review]
I'm okay with the proposal. I'm not sure what the error messages are going to be. Hope that those are very clear on what a maintainer should do.

**Review by chrisburr — APPROVED** [review]

**Review by xhochy — APPROVED** [review]

**Comment by xhochy** [comment]
The merge was accidential and shouldn't have happend. I'm very sorry for the mistake but sadly there is no simple undo here.

**Comment by ocefpaf** [comment]
> The merge was accidential and shouldn't have happend. I'm very sorry for the mistake but sadly there is no simple undo here.

This will probably pass anyway :-)

**Comment by beckermr** [comment]
Let's keep voting here. Folks should still be able to leave reviews.

**Review by h-vetinari — COMMENTED** [review]
LGTM == approval == yes (cannot approve a closed PR in GH UI)

**Comment by beckermr** [comment]
@h-vetinari i want to be sure that your lgtm is a "yes" vote. Could you not leave an approval as a review?

**Comment by beckermr** [comment]
Ahhhhh I cannot approve either. @conda-forge/core please leave your votes as a comment review on this one.

**Comment by h-vetinari** [comment]
> Ahhhhh I cannot approve either.

Yup, closed PRs cannot be approved, only reviewed. In any case, I clarified my review comment. ;-)

**Comment by mwcraig** [comment]
I approve

**Comment by loriab** [comment]
I approve

**Comment by dopplershift** [comment]
I approve

**Comment by msarahan** [comment]
approve

File: cfep-24.md
Comment (carterbox) [inline cfep-24.md:31]: I was expecting the allow-list to be more generic. For example, any feedstock with basename `foo` would be able add `foo-dev`, `foo-include`, `libfoo`, or `libfoo\d*$`. That last one especially, you're making more work for the core team whenever, for example, `libavif` bumps their soname.

Comment (beckermr) [reply cfep-24.md:31]: I think after a while the work will die down. We prepopulated the allow list by hand to help defer any initial pain. We appear to have missed `libavif` specifically.

**Comment by chenghlee** [comment]
Approve

**Comment by wolfv** [comment]
approve

**Comment by beckermr** [comment]
@conda-forge/core please vote here by leaving a comment!

**Comment by mariusvniekerk** [comment]
Approve

**Comment by bgruening** [comment]
appeove

**Comment by beckermr** [comment]
@conda-forge/core please vote!

for this one simply leave a comment since it was merged accidentally.

**Comment by beckermr** [comment]
this vote ends tomorrow Oct 31 BTW

**Comment by jezdez** [comment]
Approve

**Comment by ericdill** [comment]
Approve

**Comment by kkraus14** [comment]
Approve

File: cfep-24.md
Comment (jakirkham) [reply cfep-24.md:31]: Actually agree with Daniel here. Think we are making this too complicated. It should be safe to use some common regex expressions with the existing `<package>` and/or `<feedstock>` name

Any time a package creates a split package they will need `<package>-split` and probably several of `<package>-*` and `lib<package>*`

The split recipes are often the more involved to review. It will also unnecessarily block maintainers going through these changes while they wait for core to review

**Review by jakirkham — COMMENTED** [review]
Thanks Matt! 🙏

Appreciate you taking the time to put this together

Think I understand the motivation for this proposal and it does seem reasonable to put some more safeguards here

However as it is currently structured feel this puts undue burden on core and maintainers to validate pretty common cases (like split packages). Think we need some general regex to handle these cases

So for now unfortunately need to reject this proposal as-is with a request for revisions to address the allow-list to be more appropriately tuned. Sorry about that. Think it should be possible for me to approve with those improvements. Though would need to see them first

Reject with request for revisions

**Comment by beckermr** [comment]
Nothing in the CFEP bars us from making more general allowed patterns. The only requirement is that we have one. 

Feel free to reject, but in no way is the current implementation static IMHO.

**Comment by beckermr** [comment]
Purposely leaving the details of the name patterns out of the cfep allows us flexibility as conditions change, which is a big plus.

**Comment by beckermr** [comment]
This vote has passed. We have 25 yes, 1 no and 0 abstain. It exceeds the 60% quorum requirement of 60% (17 out of 28). Yay!

**Pull request merged**
